### DISCUSSION TITLE
🚢 New code search and code view feedback

### DISCUSSION THREAD

**Discussion by natalierjackson** [community/community#75515] [Other Feature Feedback, Questions, & Ideas]
At GitHub Universe '22  , we announced [a total redesign of GitHub's code search and navigation experience](https://github.com/features/code-search),
powered by our [all-new code search engine](https://github.blog/2023-02-06-the-technology-behind-githubs-new-code-search/) that we built from scratch. And in February 2023, we announced [our public beta](https://github.blog/changelog/2023-02-23-no-more-waitlist-code-search-and-code-view-are-available-to-all-in-public-beta/).

Today, we are rolling out this feature to all GitHub users. Thanks to the members of the beta community for your excellent feedback and engagement throughout the beta!

![Screenshot of code search results](https://user-images.githubusercontent.com/78169714/236855021-b5f60b03-78e4-46ac-a89d-7712cfb85aca.png)

Check out our [blog post](https://github.blog/2023-05-08-github-code-search-is-generally-available) to learn more about how GitHub's new code search and code view can help you search, navigate, and understand your code. And if you have feedback, please share it with us down below! 

See our older discussion post about it [here](https://github.com/orgs/community/discussions/54546)

**Comment by sixgodfan** [comment]
Search results still cannot be sorted by update time
Add Stars count filter， stars:>1000

**Comment by XDYB** [comment]
https://docs.github.com/en/search-github/github-code-search/about-github-code-search
Only UTF-8 encoded files are included?????????????
file,file,file(Game、zip、png)

**Reply by grant-h** [reply]
Yes I have this same issue. Before I was able to discover the path of binary files within repos and across the site. I don't need their contents to be indexed, just that they exist. I see that [a contributor on the last discussion mentions](https://github.com/orgs/community/discussions/55465#discussioncomment-6013777 ) that this may come in a future release. I have to say that this missing feature is a large determent to my work.

**Comment by tetrachromat** [comment]
Overall, I think the new code view is really great! However, one issue I have been running into frequently is that the code view mode will take your keyboard focus if you click on it, and any additional inputs will prompt you to open the editor- which is definitely intended, but is a bit frustrating to me. This is an event I get to experience just about daily now:

I use a keyboard-based movement plugin (vimium), so I'm relying on using my keyboard input to get around. Typically, I can use `j`/`k` to move up and down a page, and that works alright. I can also use `v` to select text typically, but this doesn't work in code view, so I guess I need to use my mouse to highlight it... and now code view has my focus. I'll press `j` to move further down the file- oh, here's a popover telling me that code view is read-only. That's right, it has my focus, so it thinks that I want to edit the file. Easy, I can press `esc` to exit the code view, and now I can move around again- but that popover is still there, reminding me that "Code view is read-only. Switch to the editor." It's blocking the line I was trying to look at! You can only clear it by clicking on code view again... which gives focus back... and the cycle repeats until I can press the keys in just the right order.

I can understand the appeal to having a way to navigate a file and copy text out by the arrow keys- I love a keyboard-driven workflow! But the interactions with keyboard focus (and especially the popover not clearing without another mouse click!) are really frustrating. I can see that it allows you to expand a selection via arrow keys, but I don't understand the need to take focus when you aren't in the editor. **I would _deeply_ appreciate a way to make the new code view's handling of keyboard focus more consistent with what users are experiencing on the rest of the web.**

An extremely simple solution might be to just close that "Code view is read-only" popover when the code view loses focus? I don't specialize in web dev, so that event may not be easy to listen for, but that would make the new code view much easier to work with. Another option might be a configurable setting for an "interactible code view". There are a lot of possible solutions that would make interacting with code view more consistent with the experience users are expecting.

**Comment by rehevkor5** [comment]
Would it be possible to add a button onto each search result which would "Exclude this repo from results"? That would make it much easier to trim down the result set while reading through it, instead of having to mentally skip past pages of irrelevant results.

**Comment by ghost** [comment]
When is this coming to GitHub Enterprise (on-prem)?

**Comment by mike-clark-8192** [comment]
I like the new code search!  I posted a piece of feedback here:
https://github.com/orgs/community/discussions/114708

**Comment by llvee** [comment]
@natalierjackson I have been discovering GH improvements. Do you have any interest in collaborating with others like myself to improve the search further? 

For example, I noticed today that topics inside of the org search allows searches that appear to search the organisation, yet actually are searching the GH website. This is technically a bug that should be resolved as the inaccurate search queries may be influencing people's engineering research efforts negatively especially those doing more programmatic research using quantitative methods.
